contracts/Refunder.sol

```diff
@@ -69,7 +76,17 @@ contract Refunder is
      */
     modifier onlySupportedParams(address targetContract, bytes4 identifier) {
         require(tx.gasprice <= maxGasPrice, "Gas price is too expensive");
-        require(refundables[targetContract][identifier], "It's not refundable");
+        require(refundables[targetContract][identifier].isSupported, "It's not refundable");
+
+        if(refundables[targetContract][identifier].validationContract != address(0)) {
+            bytes memory data = abi.encodeWithSelector(refundables[targetContract][identifier].validationFunc, msg.sender);
+            (bool success, bytes memory returnData) = refundables[targetContract][identifier].validationContract.call(data);
```

Comment:
I think that we will save gas costs if we extract:
`refundables[targetContract][identifier]` into memory variable. Test it out to see if that is the case